src/utils/generateUniqueName.ts

```diff
@@ -0,0 +1,13 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export function generateUniqueName(name: string, existingNames: string[], generateSuffix: (currentIndex: number) => string): string {
```

Comment:
This is fairly similar to `generateNonConflictFileName`, is there a way we can refactor them to share code?